File: gamedex-datamodel/src/main/java/com/github/ykrasik/gamedex/datamodel/persistence/Genre.java
package com.github.ykrasik.gamedex.datamodel.persistence;

import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.Value;

/**
 * @author Yevgeny Krasik
 */
@Value
@EqualsAndHashCode(of = "id")
public class Genre implements Comparable<Genre> {
    @NonNull private final Id<Genre> id;
    @NonNull private final String name;

    @Override
    public int compareTo(Genre o) {
        return name.compareTo(o.name);
    }

    @Override
    public String toString() {
        return name;
    }
}

File: gamedex-datamodel/src/main/kotlin/com/github/ykrasik/gamedex/datamodel/persistence/Genre.kt
package com.github.ykrasik.gamedex.datamodel.persistence

/**
 * User: ykrasik
 * Date: 25/05/2016
 * Time: 11:23
 */
data class Genre(
    val id: Int,
    val name: String
) : Comparable<Genre> {

    override fun compareTo(other: Genre) = name.compareTo(other.name)
}